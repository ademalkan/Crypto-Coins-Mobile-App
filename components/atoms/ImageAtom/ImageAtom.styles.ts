import { StyleSheet } from "react-native";

export default StyleSheet.create({
  image: {
    width: '100%',
    marginBottom: 8,
    borderRadius: 8,
  },

})