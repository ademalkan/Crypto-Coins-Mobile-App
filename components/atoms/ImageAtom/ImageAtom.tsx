import React from 'react';
import { Image, ImageSourcePropType } from 'react-native';
import styles from './ImageAtom.styles';

interface ImageAtomProps {
  image: string;
  height?: number;
}

const ImageAtom: React.FC<ImageAtomProps> = ({ image, height }) => {
  const imageSource: ImageSourcePropType = image
    ? { uri: image }
    : require('./../../../assets/images/logo.png');

  const imageStyle = height ? { ...styles.image, height: height } : styles.image;

  return <Image style={imageStyle} source={imageSource} />;
};

export default ImageAtom;
