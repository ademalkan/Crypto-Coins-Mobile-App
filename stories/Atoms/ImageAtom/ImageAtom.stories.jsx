import React from "react";
import { View } from "react-native";
import ImageAtom from "../../../components/atoms/ImageAtom/ImageAtom";

export default {
  title: "ImageAtom",
  component: ImageAtom,
};

export const Default = () => (
  <View style={{ alignItems: "center", justifyContent: "center", flex: 1 }}>
    <ImageAtom height={600} />
  </View>
);

export const SetImage = () => (
  <View style={{ alignItems: "center", justifyContent: "center", flex: 1 }}>
    <ImageAtom
      height={600}
      image="https://upload.wikimedia.org/wikipedia/commons/thumb/4/46/Bitcoin.svg/1200px-Bitcoin.svg.png"
    />
  </View>
);
