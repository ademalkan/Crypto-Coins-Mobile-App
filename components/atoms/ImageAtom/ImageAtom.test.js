import React from "react";
import renderer from "react-test-renderer";
import ImageAtom from "./ImageAtom";

jest.mock("./../../../assets/images/logo.png", () => "mocked-logo.png");

describe("ImageAtom", () => {
  it("should render the image prop correctly", () => {
    const image = "https://example.com/image.jpg";
    const component = renderer.create(<ImageAtom image={image} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render the default image when no image prop is provided", () => {
    const component = renderer.create(<ImageAtom />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render the image with custom width and height when width and height props are provided", () => {
    const image = "https://example.com/image.jpg";
    const width = 300;
    const height = 200;
    const component = renderer.create(
      <ImageAtom image={image} width={width} height={height} />
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
