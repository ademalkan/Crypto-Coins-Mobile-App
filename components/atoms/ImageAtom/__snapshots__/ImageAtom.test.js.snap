// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageAtom should render the default image when no image prop is provided 1`] = `
<Image
  source="mocked-logo.png"
  style={
    {
      "borderRadius": 8,
      "marginBottom": 8,
      "width": "100%",
    }
  }
/>
`;

exports[`ImageAtom should render the image prop correctly 1`] = `
<Image
  source={
    {
      "uri": "https://example.com/image.jpg",
    }
  }
  style={
    {
      "borderRadius": 8,
      "marginBottom": 8,
      "width": "100%",
    }
  }
/>
`;

exports[`ImageAtom should render the image with custom height when height prop is provided 1`] = `
<Image
  source={
    {
      "uri": "https://example.com/image.jpg",
    }
  }
  style={
    {
      "borderRadius": 8,
      "height": 200,
      "marginBottom": 8,
      "width": "100%",
    }
  }
/>
`;
